src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1536,24 +1534,15 @@ void TorrentHandle::renameFile(const int index, const QString &name)
 bool TorrentHandle::saveTorrentFile(const QString &path)
 {
     if (!m_torrentInfo.isValid()) return false;
-#if (LIBTORRENT_VERSION_NUM < 10200)
-    const lt::create_torrent torrentCreator = lt::create_torrent(*(m_torrentInfo.nativeInfo()), true);
-#else
-    const lt::create_torrent torrentCreator = lt::create_torrent(*(m_torrentInfo.nativeInfo()));
-#endif
-    const lt::entry torrentEntry = torrentCreator.generate();
 
-    QByteArray out;
-    out.reserve(1024 * 1024);  // most torrent file sizes are under 1 MB
-    lt::bencode(std::back_inserter(out), torrentEntry);
-    if (out.isEmpty())
+    try {
+        m_torrentInfo.saveToFile(path);
+    }
+    catch (...) {
```

Comment:
I think you should catch `RuntimeError` exceptions only.

Reply:
Also, shouldn't you write to log if it fails?

Reply:
There was no logging in this method before....
Also I would prefer writing to the log in more meaningful context, i.e. when we trigger file saving and can provide more info.
IMO, it's better to pass the exception to upper level and handle it there. Bit it's a job for separate PR.

Reply:
>But it's a job for separate PR.

Still done here.